### PR TITLE
cn: added node/cn/backend_test.go

### PR DESCRIPTION
**Pull request by ehnuje**
## Proposed changes

- cn: added node/cn/backend_test.go

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [x] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: node/cn/backend_test.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+	setEngineType(cc)
+	assert.Equal(t, types.Engine_Clique, types.EngineType)
+
+	cc.Istanbul = &params.IstanbulConfig{}
+	setEngineType(cc)
+	assert.Equal(t, types.Engine_IBFT, types.EngineType)
+}
+
+func TestCN_SetAcceptTxs(t *testing.T) {
+	{
```
Comment (jeonghonah) [inline node/cn/backend_test.go]:
```suggestion
```

File: node/cn/backend_test.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	mockBlockChain := mocks2.NewMockBlockChain(mockCtrl)
+	cn.blockchain = mockBlockChain
+	block := blocks[0]
+
+	mockBlockChain.EXPECT().ResetWithGenesisBlock(block).Times(1)
+	cn.ResetWithGenesisBlock(block)
+}
+
+func TestCN_Rewardbase(t *testing.T) {
+	{
```
Comment (jeonghonah) [inline node/cn/backend_test.go]:
```suggestion
```

File: node/cn/backend_test.go (hunk 1 above)
Comment (KimKyungup) [reply node/cn/backend_test.go]: @melvin Will you add another test case in next PR for covering below code?
```
// setAcceptTxs sets AcceptTxs flag in 1CN case to receive tx propagation.
func (s *CN) setAcceptTxs() error {
	if s.chainConfig.Istanbul != nil {
		istanbulExtra, err := types.ExtractIstanbulExtra(s.blockchain.Genesis().Header())
		if err != nil {
			return err
		} else {
			if len(istanbulExtra.Validators) == 1 {
				atomic.StoreUint32(&s.protocolManager.acceptTxs, 1)
			}
		}
	}
	return nil
}```

File: node/cn/backend_test.go (hunk 2 above)
Comment (KimKyungup) [reply node/cn/backend_test.go]: @melvin Will you add another test cases in next PR?

File: node/cn/backend_test.go (hunk 1 above)
Comment (ehnuje) [reply node/cn/backend_test.go]: @KimKyungup Yes, I will. Currently this test covers only a part of the function. But properly setting `ExtractIstanbulExtra` is needed to extend the coverage. I will update the test in another PR. cc @jeonghonah

File: node/cn/backend_test.go (hunk 2 above)
Comment (ehnuje) [reply node/cn/backend_test.go]: Same comment as above. To extend the coverage, mocking `accounts.Manager` is needed.

**Comment by ehnuje** [comment]
@jeonghonah Could you please review this change?

**Pull request merged**